### PR TITLE
Issue 191

### LINKED ISSUE #191: _NET_CURRENT_DESKTOP not always updated when new workspace is focused
### Describe the bug

The _NET_CURRENT_DESKTOP is not always updated when a new workspace is focused.
This causes programs such as polybar to display bad information to the user.

### To Reproduce

The places where this behavior is noticed:
 - when cycling screens with `cycle_screen`
 - `focus_workspace` when the new workspace was on a different screen
 - dragging the mouse to a new screen, where this workspace is focused

### Expected behavior

Every time a new workspace is focused, the property should be updated.

### Versions & OS Details

- OS: linux
- Distribution: Arch
- Penrose Version: cc36bea96eb9863a9c0ad42f42fe8c7cc2c78571

### Additional context

I created a fork with a temporary patch, calling `self.conn.set_current_workspace` several additional places in the `WindowManager`.
This should probably be abstracted out and put in one place so one does not have to remember to call this every time a new workspace is focused.

### PR DESCRIPTION
**Pull request by teo8192**
Resolves #191

### CODE REVIEW

File: src/core/manager/mod.rs [hunk 1]
```diff
@@ -782,6 +792,10 @@ impl<X: XConn> WindowManager<X> {
     /// Cycle between known [screens][Screen]. Does not wrap from first to last
     pub fn cycle_screen(&mut self, direction: Direction) -> Result<()> {
         let actions = self.state.screens.cycle_screen(direction, &self.conn)?;
+
+        let wix = self.screens.active_ws_index();
+        self.conn.set_current_workspace(wix)?;
```
Comment (sminez) [inline src/core/manager/mod.rs]: This is redundant as it you are already handling this in `focus_workspace` which will be called as a result of the event actions here

File: src/core/manager/mod.rs [hunk 2]
```diff
@@ -726,11 +726,21 @@ impl<X: XConn> WindowManager<X> {
...
     fn focus_screen(&mut self, sel: &Selector<'_, Screen>) -> &Screen {
+        let prev_wix = self.screens.focused().wix;
+
         let actions = self.screens.focus_screen(sel);
         if let Err(e) = self.handle_event_actions(actions) {
             (self.error_handler)(e);
         }
 
+        // update current workspace if it is different
+        let wix = self.screens.focused().wix;
```
Comment (sminez) [inline src/core/manager/mod.rs:737]: same here

**Review by sminez — CHANGES_REQUESTED** [review]
Can you please add some tests to this ensure that the workspace is only set once when changing screen or workspace. It looks the current code is going to trigger the call to the X server multiple times.

File: src/core/manager/mod.rs (hunk 1 above)
Comment (teo8192) [reply src/core/manager/mod.rs]: I think the `Screens::cycle_screen` changes the screen, causing `focus_workspace` to exit early in the first if-test and not handling it. I will try to write some tests to confirm this and not having to rely on my manual testing.

File: src/core/manager/mod.rs (hunk 2 above)
Comment (teo8192) [reply src/core/manager/mod.rs:737]: The event action also stops at the first if-test in `focus_workspace` here.

**Review by sminez — APPROVED** [review]

**Pull request merged**